qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -160,32 +161,47 @@ def _component_data(self, composite_data: List[Dict]) -> List[List[Dict]]:
         # Sort by index
         return [marginalized_data[i] for i in sorted(marginalized_data.keys())]
 
-    def _initialize_components(self, experiment_data: ExperimentData) -> List[str]:
-        """Initialize child data components and return list of child experiment IDs"""
-        # Check if component child experiment data containers have already
-        # been created. If so the list of indices for their positions in the
-        # ordered dict should exist. Index is used to extract the experiment
-        # IDs for each child experiment which can change when re-running analysis
-        # if replace_results=False, so that we update the correct child data
-        # for each component experiment
+    def _initialize_child_data(self, experiment_data: ExperimentData):
+        """Initialize component experiment data objects as child data"""
         component_index = experiment_data.metadata.get("component_child_index", [])
-        if not component_index:
-            experiment = experiment_data.experiment
-            if experiment is None:
-                raise AnalysisError(
-                    "Cannot run composite analysis on an experiment data without either "
-                    "a composite experiment, or composite experiment metadata."
-                )
-            # If the experiment Construct component data and update indices
-            start_index = len(experiment_data.child_data())
-            component_index = []
-            for i, sub_exp in enumerate(experiment.component_experiment()):
-                sub_data = sub_exp._initialize_experiment_data()
-                experiment_data.add_child_data(sub_data)
-                component_index.append(start_index + i)
-            experiment_data.metadata["component_child_index"] = component_index
-
-        # Child components exist so we can get their ID for accessing them
-        child_ids = experiment_data._child_data.keys()
-        component_ids = [child_ids[idx] for idx in component_index]
-        return component_ids
+        if component_index:
+            # Child components are already initialized
+            return
+
+        # Initialize the component experiment data containers and add them
+        # as child data to the current experiment data
+        child_components = self._initialize_component_data(experiment_data)
+        start_index = len(experiment_data.child_data())
+        for i, subdata in enumerate(child_components):
+            experiment_data.add_child_data(subdata)
+            component_index.append(start_index + i)
+
+        # Store the indices of the added child data in metadata
+        experiment_data.metadata["component_child_index"] = component_index
+
+    def _initialize_component_data(self, experiment_data: ExperimentData) -> List[ExperimentData]:
```

Comment:
Having `initialize_component_data` and `initialize_child_data` are really confusing for developers. As far as I can understand, there is no physical `component_data` member in this class (indeed this is deprecated attribute), and what this method is doing looks like initializing the data container for the _experiment data of child experiment_. What about renaming this to `_create_empty_containers`? Probably removing this method and just merging the logic here to `_initialize_child_data` is also fine.